### PR TITLE
Adding integration test cases for filedatabase service

### PR DESCRIPTION
**Pull request by anandmohan777**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +26,7 @@ $ mvn test
 The default value for properties are written is pom.xml inside `<systemPropertyVariables>` for example the key server host is assumed to be running on `http://localhost`. To override this value at run time use below command
 
 ```
-$ mvn test "-Dkey.uri=http://localhost2"
+$ mvn test "-Dkey.host=http://localhost2"
```
Comment (omllobet) [inline README.md]: Technically still a url :)

File: src/main/java/eu/elixir/ega/ebi/it/filedatabase/FileDatabaseBase.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import com.jayway.restassured.specification.RequestSpecification;
+
+public class FileDatabaseBase {
+
+    final static String ID_ZERO = "000";
+    RequestSpecification REQUEST;
+    String datasetId;
+    String fileId;
+    String indexId;
+    String keyId;
```
Comment (omllobet) [inline src/main/java/eu/elixir/ega/ebi/it/filedatabase/FileDatabaseBase.java]: keyId seems to be not used anywhere

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+    }
+    
+    /**
+     * Verify the api call /datasets/{fileId}/files and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetDatasetFilesForZeroId() {
+        final Response response = REQUEST.get(DATASET_CONTROLLER +  ID_ZERO + "/files");
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java:45]: To me it seems an strange api design that the dataset just return empty if you pass a dataset that does not exist

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * 
+ * @author amohan
+ *
+ */
+public class DatasetControllerIntgTTest extends FileDatabaseBase {
+
+    private final static String DATASET_CONTROLLER = "/datasets/";    
+    
+    /**
+     * Verify the api call /datasets/{fileId}/files and check status is {@link org.apache.http.HttpStatus#SC_OK}.
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java]: /datasets/{datasetId}/files

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(1));
+        assertThat(jsonObject.getString("datasetId"), equalTo(datasetId));
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+    }
+    
+    /**
+     * Verify the api call /datasets/{fileId}/files and check status is {@link org.apache.http.HttpStatus#SC_OK}.
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java]: /datasets/{datasetId}/files

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        assertThat(jsonArray.length(), equalTo(1));
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId} and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileForZeroId() {
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:43]: The same comment as for the non existing dataset

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        assertThat(jsonObject.getString("datasetId"), equalTo(datasetId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId}/datasets and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileDatasetForZeroId() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + ID_ZERO + "/datasets");
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:74]: The same comment as for the non existing dataset

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        assertThat(jsonObject.getString("indexFileId"), equalTo(indexId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId}/index and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileIndexForZeroId() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + ID_ZERO + "/index");
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:103]: The same comment as for the non existing dataset

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(0));
+    }
+
+}
```
Comment (omllobet) [inline src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:110]: Would it be possible to add some checks on exceptions? Probably some conditions should make the data get file.

**Review by omllobet — CHANGES_REQUESTED** [review]
I have not tried to run the tests themselves yet. I'll try the sql file and run everything and let you know.
Maybe now with the .sql file, it would be a good time to start adding travis integration and run this test like everyday or after a commit in ega-data-api?

Also please rebase on top of master

File: README.md (hunk 1 above)
Comment (anandmohan777) [reply README.md]: changed to key.url

File: src/main/java/eu/elixir/ega/ebi/it/filedatabase/FileDatabaseBase.java (hunk 2 above)
Comment (anandmohan777) [reply src/main/java/eu/elixir/ega/ebi/it/filedatabase/FileDatabaseBase.java]: done

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java (hunk 3 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java:45]: Yes, currently it is implemented like this. It returns empty result array if id not found in the database.

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java (hunk 4 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java]: Done

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java (hunk 5 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/DatasetControllerIntgTTest.java]: Done

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java (hunk 6 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:43]: Yes, currently it is implemented like this. It returns empty result array if id not found in the database.

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java (hunk 7 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:74]: Yes, currently it is implemented like this. It returns empty result array if id not found in the database.

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java (hunk 8 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:103]: Yes, currently it is implemented like this. It returns empty result array if id not found in the database.

File: src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java (hunk 9 above)
Comment (anandmohan777) [reply src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java:110]: In the application implementation class (FileServiceImpl), there is no exception thrown in any condition. So there is no scenario to test the exceptions.

**Review by omllobet — APPROVED** [review]

**Comment by omllobet** [comment]
I also tested it and the tests passed

**Pull request merged**